packages/frontend/src/state/positions/providers.tsx

```diff
@@ -0,0 +1,110 @@
+          const squeethAmt = new BigNumber(isWethToken0 ? s.amount1 : s.amount0)
+          const wethAmt = new BigNumber(isWethToken0 ? s.amount0 : s.amount1)
+          const usdAmt = getUsdAmt(wethAmt, s.timestamp)
+          //buy one squeeth means -1 to the pool, +1 to the user
+          acc.squeethAmount = acc.squeethAmount.plus(squeethAmt.negated())
+          //<0 means, buying squeeth
+          //>0 means selling squeeth
+          if (squeethAmt.isPositive()) {
+            //sold Squeeth amount
+            acc.soldSqueeth = acc.soldSqueeth.plus(squeethAmt.abs())
+            //usd value from sell to close long position or open short
+            acc.totalUSDFromSell = acc.totalUSDFromSell.plus(usdAmt.abs())
+          } else if (squeethAmt.isNegative()) {
+            //bought Squeeth amount
+            acc.boughtSqueeth = acc.boughtSqueeth.plus(squeethAmt.abs())
+            //usd value from buy to close short position or open long
+            acc.totalUSDFromBuy = acc.totalUSDFromBuy.plus(usdAmt.abs())
+          }
+          if (acc.squeethAmount.isZero()) {
+            acc.longUsdAmount = BIG_ZERO
+            acc.shortUsdAmount = BIG_ZERO
+            acc.wethAmount = BIG_ZERO
+            acc.boughtSqueeth = BIG_ZERO
+            acc.soldSqueeth = BIG_ZERO
+            acc.totalUSDFromSell = BIG_ZERO
+            acc.totalUSDFromBuy = BIG_ZERO
+          } else {
+            // when the position is partially closed, will accumulate usdamount
+            acc.longUsdAmount = acc.longUsdAmount.plus(usdAmt)
+            acc.shortUsdAmount = acc.shortUsdAmount.plus(usdAmt.negated())
+            acc.wethAmount = acc.wethAmount.plus(wethAmt.negated())
+          }
+          return acc
+        },
+        {
+          squeethAmount: BIG_ZERO,
+          wethAmount: BIG_ZERO,
+          longUsdAmount: BIG_ZERO,
+          shortUsdAmount: BIG_ZERO,
+          boughtSqueeth: BIG_ZERO,
+          soldSqueeth: BIG_ZERO,
+          totalUSDFromBuy: BIG_ZERO,
+          totalUSDFromSell: BIG_ZERO,
+        },
+      ) || {
+        squeethAmount: BIG_ZERO,
+        wethAmount: BIG_ZERO,
+        longUsdAmount: BIG_ZERO,
+        shortUsdAmount: BIG_ZERO,
+        boughtSqueeth: BIG_ZERO,
+        soldSqueeth: BIG_ZERO,
+        totalUSDFromBuy: BIG_ZERO,
+        totalUSDFromSell: BIG_ZERO,
+      },
+    [isWethToken0, data?.swaps, getUsdAmt],
+  )
+
+  useEffect(() => {
+    if (computedSwaps.squeethAmount.isGreaterThan(0)) {
+      setPositionType(PositionType.LONG)
+    } else if (computedSwaps.squeethAmount.isLessThan(0)) {
+      setPositionType(PositionType.SHORT)
+    } else setPositionType(PositionType.NONE)
+  }, [computedSwaps.squeethAmount, setPositionType])
+
+  const value = useMemo(
+    () => ({ ...computedSwaps, squeethAmount: computedSwaps.squeethAmount.absoluteValue() }),
+    [computedSwaps],
+  )
+
+  return <ComputeSwapsContext.Provider value={value}>{children}</ComputeSwapsContext.Provider>
+}
```

Comment:
@adam5909099 noticed we replaced the useComputeSwaps hook with context, please can you give some context(no pun intended😂) on your thoughts behind that

Reply:
useComputeSwaps is used in more than 10 places and the useMemo was called separately in each place. So I moved it into context to make sure it's called one time in global level.